llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -29318,8 +29318,10 @@ static SDValue LowerShift(SDValue Op, const X86Subtarget &Subtarget,
   if (SDValue V = LowerShiftByScalarImmediate(Op, DAG, Subtarget))
     return V;
 
-  if (SDValue V = LowerShiftByScalarVariable(Op, DAG, Subtarget))
-    return V;
+  if (!supportedVectorVarShift(VT, Subtarget, Opc) &&
+      !Subtarget.preferPerEltVectorShift())
```

Comment:
This logic looks wrong - shouldn't it be `!supportedVectorVarShift(VT, Subtarget, Opc) || !Subtarget.preferPerEltVectorShift()`?